src/design-pages/passkey_kerberos.rst

```diff
@@ -0,0 +1,223 @@
+Passkey authentication Kerberos integration
+=======================================================
+
+Related ticket(s):
+------------------
+ * https://github.com/SSSD/sssd/issues/6228
+
+Problem statement
+-----------------
+This design is an extension of the "enable passkey authentication", where a
+centrally managed user can authenticate locally in a system with a passkey
+token. This might not be enough for SSSD in some environments like big
+companies and governments, thus it’s necessary to add an extension to provide
+a Kerberos ticket while authenticating the user. This way the users will be
+able to identify themselves to other services, which means that they can use
+the Kerberos ticket for Single Sign-on (SSO).
+
+For the purpose of this feature, passkey is a FIDO2 compatible device supported
+by the libfido2 library.
+
+Use cases
+---------
+As a user, not only do I want to walk to any system on a network and log in
+(desktop and/or terminal) using a passkey device connected to the system, but I
+also want to get a Kerberos ticket to identify myself to other services.
+
+Prerequisites
+-------------
+To register a new
+`pre-authentication data identifier <https://www.iana.org/assignments/kerberos-parameters/kerberos-parameters.xhtml#pre-authentication>`__
+in Kerberos for passkey it would be good to first finalize the IdP plugin
+design and security analysis requested by this community.
+
+Solution overview
+-----------------
+The objective is to locally authenticate and issue a Kerberos ticket to a
+centrally managed user. For that purpose SSSD will need to be modified to
+accommodate the new workflow, while still allowing the old one, where the user
+was authenticated locally without issuing any Kerberos ticket. The new workflow
+will be used for online authentication against an IPA server. The old workflow
+will be used for any offline authentication and for AD and other LDAP servers.
+
+SSSD will send a pre_auth message to the Kerberos client. It will send an
```

Comment:
... "SSSD provides a pre-authentication method implementation (pre-auth plugin) to MIT Kerberos library. When both the Kerberos client and the KDC support the same pre-authentication method, KDC will respond to an AS-REQ message with a request to provide an assertion data required by a pre-authentication method. SSSD's pre-auth plugin will obtain the assertion data from the passkey. This information will be sent by the Kebreros client back to the KDC server as a part of the AS-REQ exchange."

Reply:
Updated